### ISSUE TITLE
response-content-disposition as a query param does not work because of spaces not escaped properly

### ISSUE THREAD

**Issue by cfactolerin** [googleapis/google-cloud-ruby#7655]
#### Environment details

  - OS: Mac OS Catalina v10.15.3
  - Ruby version: ruby 2.6.5p114
  - Gem name and version: google-cloud-storage

#### Steps to reproduce

  1. Pass `"response-content-disposition" => "attachment; filename=\"test 1.txt\""` as a query parameter with a file name that has space in between. 
 2. You will get a `SignatureDoesNotMatchThe request signature we calculated does not match the signature you provided error.`

#### Code example

```ruby
storage = Google::Cloud::Storage.new(project_id: 'xxx-project-id, credentials: 'credentials_path/config.json')
bucket = storage.bucket('my-bucket-name)
bucket.signed_url('filepath/filename.txt', version: :v4, expires: 3600, headers: nil, query: {"response-content-disposition"=>"attachment; filename=\"test 1.txt\""}, method: :GET)
```

#### Full backtrace

No back trace because it still provides you a signed url. It's just that when you use the url you will seem something like:
```
SignatureDoesNotMatchThe request signature we calculated does not match the signature you provided. Check your Google secret key and signing method.
```

The issue is that CGI.escape converts the spaces into '+' which is not the  %XX UTF-8 style that you guys expected.
Your test cases pass because you guys used URI to parse the url. URI uses %20 for spaces.

I did a quick fork on the master branch to try out fixing it and it works. Do have a look and let me know if I should do a PR for it. https://github.com/cfactolerin/google-cloud-ruby/commit/90cf55c870389b67df3da8741ee403e22dc1c669

Thanks!

**Comment by quartzmo** [comment]
@cfactolerin Thank you for reporting this issue. I'm checking with some product folks for guidance on the fix, but most likely it will be great to get your changes in a PR.

**Comment by quartzmo** [comment]
@cfactolerin Please open a PR, thanks in advance!

**Comment by cfactolerin** [comment]
Done! Thanks 😄

**Comment by Achooo** [comment]
Hey! @cfactolerin Thank you for this, I am facing a similar issue 🙏 

@quartzmo A quick question about V4 signing, for V2 signed urls we were able to **append** a correctly encoded `response-content-disposition`. We tried doing the same for V4 urls to get around this problem, however it does not seem to be working.

Out of curiosity for V4 signed urls, do query parameters such as `response-content-disposition` impact/play a role in the signature generation?
